### PR TITLE
[MNT-22353]- Removed Hold cases to be processed in DispositionLifecycleJob

### PR DESCRIPTION
**Pull request by pjoshi31**

### CODE REVIEW

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java [hunk 1]
```diff
@@ -265,12 +268,6 @@ private void executeAction(final List<NodeRef> actionNodes)
                 }
                 Map<String, Serializable> props = Map.of(PARAM_NO_ERROR_CHECK, false);
 
-                if (freezeService.isFrozenOrHasFrozenChildren(parent.getParentRef()))
-                {
```
Comment (kavitshah-gl) [inline amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java:269]: Why is the log for the debug removed? This should be present to see what is the value during the debugging.

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java [hunk 2]
```diff
@@ -214,8 +215,10 @@ public void executeImpl()
 
                 // execute search
                 ResultSet results = searchService.query(params);
-                List<NodeRef> resultNodes = results.getNodeRefs();
                 hasMore = results.hasMore();
+                if(results.length() != 0) {
+                    resultNodes=results.getNodeRefs().stream().filter(node ->!freezeService.isFrozenOrHasFrozenChildren(nodeService.getPrimaryParent(node).getParentRef())).collect(Collectors.toList());
```
Comment (kavitshah-gl) [inline amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java:220]: Put the comments as what is these changes doing.

**Review by kavitshah-gl — CHANGES_REQUESTED** [review]

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java (hunk 2 above)
Comment (pjoshi31) [reply amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java:221]: Added

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java (hunk 1 above)
Comment (pjoshi31) [reply amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java:269]: In this If condition there will be not frozen cases, this condition is of no use then.

**Review by kavitshah-gl — APPROVED** [review]
Approved.

File: amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java [hunk 3]
```diff
@@ -214,7 +215,14 @@ public void executeImpl()
 
                 // execute search
                 ResultSet results = searchService.query(params);
-                List<NodeRef> resultNodes = results.getNodeRefs();
+                if(results != null)
+                {
+                    // filtering out the hold/freezed cases from the result set
+                    resultNodes =
+                            results.getNodeRefs().stream().filter(node -> nodeService.getPrimaryParent(node) == null ?
```
Comment (tpage-alfresco) [inline amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java:222]: How is it possible for `nodeService.getPrimaryParent(node) == null`?

Comment (pjoshi31) [reply amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/job/DispositionLifecycleJobExecuter.java:222]: I'm not aware of the scenario but when we are removing the null check then the Test cases are getting failed with HeapMemory Issue.

**Pull request merged**